cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueDebugger.java

```diff
@@ -0,0 +1,280 @@
+   * @param stateValue the value of the state column in the row
+   * @param queueRowPrefixLength length of the queueRowPrefix
+   */
+  private void visitRow(QueueStatistics stats, Transaction tx, byte[] rowKey,
+                        byte[] stateValue, int queueRowPrefixLength) {
+    if (stateValue == null) {
+      stats.countUnprocessed(1);
+      return;
+    }
+
+    ConsumerEntryState state = QueueEntryRow.getState(stateValue);
+    if (state == ConsumerEntryState.PROCESSED) {
+      long writePointer = QueueEntryRow.getWritePointer(rowKey, queueRowPrefixLength);
+      if (tx.isVisible(writePointer)) {
+        stats.countProcessedAndVisible(1);
+      } else {
+        stats.countProcessedAndNotVisible(1);
+      }
+    }
+  }
+
+  /**
+   *
+   */
+  private static final class QueueStatistics {
+
+    private long unprocessed;
+    private long processedAndVisible;
+    private long processedAndNotVisible;
+
+    private QueueStatistics() {
+    }
+
+    public void countUnprocessed(long count) {
+      unprocessed += count;
+    }
+
+    public void countProcessedAndVisible(long count) {
+      processedAndVisible += count;
+    }
+
+    public void countProcessedAndNotVisible(long count) {
+      processedAndNotVisible += count;
+    }
+
+    public long getUnprocessed() {
+      return unprocessed;
+    }
+
+    public long getProcessedAndVisible() {
+      return processedAndVisible;
+    }
+
+    public long getProcessedAndNotVisible() {
+      return processedAndNotVisible;
+    }
+
+    public long getTotal() {
+      return unprocessed + processedAndVisible + processedAndNotVisible;
+    }
+
+    public String getReport() {
+      return String.format("unprocessed: %d; processed and visible: %d; processed and not visible: %d; total: %d",
+                           getUnprocessed(), getProcessedAndVisible(), getProcessedAndNotVisible(), getTotal());
+    }
+  }
+
+  public static void main(String[] args) throws Exception {
+    Injector injector = Guice.createInjector(
+      new ConfigModule(),
+      new ZKClientModule(),
+      new TransactionClientModule(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          bind(QueueClientFactory.class).to(HBaseQueueClientFactory.class).in(Singleton.class);
+          bind(QueueAdmin.class).to(HBaseQueueAdmin.class).in(Singleton.class);
+          bind(HBaseTableUtil.class).toProvider(HBaseTableUtilFactory.class);
+        }
+      },
+      new DiscoveryRuntimeModule().getDistributedModules(),
+      new LocationRuntimeModule().getDistributedModules(),
+      new DataSetsModules().getDistributedModules(),
+      new TransactionModules().getDistributedModules()
+    );
+
+    HBaseQueueDebugger debugger = injector.getInstance(HBaseQueueDebugger.class);
+    debugger.startAndWait();
+
+    // e.g. "queue:///default/PurchaseHistory/PurchaseFlow/reader/queue"
+    final QueueName queueName = QueueName.from(URI.create(args[0]));
```

Comment:
At least mark it as a TODO. For now, if the args.length == 0, prints a help message to tell how to construct the queue name URI.